sdk/src/tx/whileValidTxSender.ts

```diff
@@ -0,0 +1,208 @@
+
+		return signedTx;
+	}
+
+	async getVersionedTransaction(
+		ixs: TransactionInstruction[],
+		lookupTableAccounts: AddressLookupTableAccount[],
+		_additionalSigners?: Array<Signer>,
+		_opts?: ConfirmOptions
+	): Promise<VersionedTransaction> {
+		const message = new TransactionMessage({
+			payerKey: this.wallet.publicKey,
+			recentBlockhash: '', // set blank and reset in sendVersionTransaction
+			instructions: ixs,
+		}).compileToV0Message(lookupTableAccounts);
+
+		const tx = new VersionedTransaction(message);
+
+		return tx;
+	}
+
+	async sendVersionedTransaction(
+		tx: VersionedTransaction,
+		additionalSigners?: Array<Signer>,
+		opts?: ConfirmOptions,
+		preSigned?: boolean,
+		extraConfirmationOptions?: ExtraConfirmationOptions
+	): Promise<TxSigAndSlot> {
+		const latestBlockhash = await this.connection.getLatestBlockhash();
+		tx.message.recentBlockhash = latestBlockhash.blockhash;
+
+		let signedTx;
+		if (preSigned) {
+			signedTx = tx;
+			// @ts-ignore
+		} else if (this.wallet.payer) {
+			// @ts-ignore
+			tx.sign((additionalSigners ?? []).concat(this.wallet.payer));
+			signedTx = tx;
+		} else {
+			additionalSigners
+				?.filter((s): s is Signer => s !== undefined)
+				.forEach((kp) => {
+					tx.sign([kp]);
+				});
+			// @ts-ignore
+			signedTx = await this.wallet.signTransaction(tx);
+		}
+
+		if (extraConfirmationOptions?.onSignedCb) {
+			extraConfirmationOptions.onSignedCb();
+		}
+
+		if (opts === undefined) {
+			opts = this.opts;
+		}
+
+		const txSig = bs58.encode(signedTx.signatures[0]);
+		this.untilValid.set(txSig, latestBlockhash);
+
+		return this.sendRawTransaction(signedTx.serialize(), opts);
+	}
+
+	async sendRawTransaction(
+		rawTransaction: Buffer | Uint8Array,
+		opts: ConfirmOptions
+	): Promise<TxSigAndSlot> {
+		const startTime = this.getTimestamp();
+
+		const txid = await this.connection.sendRawTransaction(rawTransaction, opts);
+		this.sendToAdditionalConnections(rawTransaction, opts);
+
+		let done = false;
+		const resolveReference: ResolveReference = {
+			resolve: undefined,
+		};
+		const stopWaiting = () => {
+			done = true;
+			if (resolveReference.resolve) {
+				resolveReference.resolve();
+			}
+		};
+
+		(async () => {
+			while (!done && this.getTimestamp() - startTime < this.timeout) {
+				await this.sleep(resolveReference);
+				if (!done) {
+					this.connection
+						.sendRawTransaction(rawTransaction, opts)
+						.catch((e) => {
+							console.error(e);
+							stopWaiting();
+						});
+					this.sendToAdditionalConnections(rawTransaction, opts);
+				}
+			}
+		})();
+
+		let slot: number;
+		try {
+			const { blockhash, lastValidBlockHeight } = this.untilValid.get(txid);
+			const result = await this.connection.confirmTransaction(
+				{
+					signature: txid,
+					lastValidBlockHeight,
+					blockhash,
+				},
+				opts.commitment
+			);
+			slot = result.context.slot;
+			// eslint-disable-next-line no-useless-catch
+		} catch (e) {
+			throw e;
```

Comment:
`confirmTransaction` throws if the tx expires, do we want that to throw here too? client will have to make sure it's handled properly